build: Make 'prettier -w .' work from the root

tylerbutler opened the pull request:
Related to #15758.

This PR adds new prettier scripts to run on the whole repo. I also updated the prettierignore files to ignore additional files. The `prettier --write .` command should also now work from the root of any release group.

tylerbutler committed:
build: Make 'prettier -w .' work

tylerbutler committed:
updates

tylerbutler committed:
enable docs

tylerbutler committed:
build: Update shared prettier config

tylerbutler committed:
updates

tylerbutler committed:
updates

tylerbutler committed:
merge

tylerbutler committed:
revert

alexvy86 commented on .prettierignore:
```diff
@@ -70,3 +70,24 @@ packages/framework/data-object-base/es5
+# These files are auto-generated according to the comments in the files
+**/charts/**/Chart.yaml
+
+# Kubernetes
+**/kubernetes/**
```
Any particular reason for this one? I've dealt with those files (/server/routerlicious/kubernetes) and I think the yaml and markdown ones wouldn't mind auto-formatting (templates excluded, but that's handled above already).

tylerbutler replied:
I'll double check this.

scottn12 committed:
Automation: Main Next Integrate (#15817)

scottn12 committed:
Automation: Main Next Integrate (#15821)

markfields committed:
Add support for logging flat objects (e.g. details) to reduce number of unique columns in telemetry (#15827)

## Description

Precursors:
* PR https://github.com/microsoft/FluidFramework/pull/12863 moved the
typings for our internal loggers to telemetry-utils package and added
support for arrays.
* PR #15667 upgraded the client release group to use the new types

This PR updates that type further to add support for flat objects. This
has become a more common practice to consolidate event-specific
properties, e.g. in a `details` object. Today instrumentation does the
JSON.stringify itself, but this is not ideal. It's proven the utility of
logging properties like this, so this PR adds first-class support for
this.

## Breaking Changes

Yes, this is a breaking change. If a logger implements the existing
logger contracts based on the existing type, it will not necessarily be
equipped to handle a flat object.

In practice, there shouldn't be any follow-up required. These types are
only used for our internal logging contracts, which are always fulfilled
by a class derived from `TelemetryLogger`, which actually handles this
case just fine without any changes.

The API surface we expect partners to use always speaks in
`ITelemetryBaseLogger` which is unchanged here.

tylerbutler committed:
rm markdown overrides

tylerbutler committed:
Merge branch 'main' into prettier-config

tylerbutler committed:
formatting

tylerbutler committed:
Merge branch 'prettier-config' into docs-prettier

tylerbutler committed:
settings

tylerbutler committed:
run format from the root

alexvy86 commented on package.json:
```diff
@@ -45,23 +45,21 @@
-		"prettier:fix": "npm run prettier:root:fix && pnpm run -r --no-sort --stream prettier:fix",
-		"prettier:root": "prettier --check ./*.* --ignore-path .prettierignore",
-		"prettier:root:fix": "prettier --write ./*.* --ignore-path .prettierignore",
+		"prettier": "prettier --check . --ignore-path ./.prettierignore",
+		"prettier:fix": "prettier --write . --ignore-path ./.prettierignore",
```
This will this start formatting the server release group when run from the root, right? Is that intended? I think it shouldn't...

tylerbutler replied:
Yes, it's intended. In the ideal case, this should never be a problem because CI enforces that everything is formatted properly already, so 

If we don't do it this way, where should the script to format the whole repo live?

alexvy86 replied:
I think my honest answer to that is "here but server should live in a separate repo" 😆. Not strongly opposed to moving forward like this, just noting that my spidey sense reacted. One specific concern I have is if server folks update "their copy" of `.prettierignore` but no the root; we expect CI to fail in that case because it will format from the root and cause the validation for unexpected file changes to complain?

tylerbutler replied:
I created new prettier:repo and format:repo scripts so the old ones still stay scoped to the client release group.

tylerbutler committed:
Merge branch 'main' into docs-prettier

tylerbutler committed:
update

tylerbutler commented on experimental/dds/tree-graphql/package.json:
```diff
@@ -28,8 +28,8 @@
 		"lint:fix": "npm run prettier:fix && npm run eslint:fix",
-		"prettier": "prettier --check \"**/*.{js,json,jsx,md,ts,tsx,yml,yaml}\" --ignore-path ../../../.prettierignore",
-		"prettier:fix": "prettier --write \"**/*.{js,json,jsx,md,ts,tsx,yml,yaml}\" --ignore-path ../../../.prettierignore",
+		"prettier": "prettier --check . --ignore-path ../../../.prettierignore",
+		"prettier:fix": "prettier --write . --ignore-path ../../../.prettierignore",
```
Yeah.

tylerbutler committed:
Merge branch 'main' into docs-prettier

tylerbutler committed:
formatting

tylerbutler committed:
Merge branch 'main' into docs-prettier

tylerbutler committed:
Automation: Main Next Integrate

alexvy86 reviewed (approved):
PR description maybe needs an update, but the change LGTM.

tylerbutler committed:
Merge branch 'main' into docs-prettier

tylerbutler committed:
ignorefiles

tylerbutler committed:
Merge branch 'main' into docs-prettier
